### PR TITLE
move dtogen out of utils repo

### PR DESCRIPTION
**Pull request by tvannoy**
dtogen was taken from the experimentDTO branch of the utils repo. @fe-wickham if that isn't the branch we should be using, we can update this.

### CODE REVIEW

File: device_tree_overlays/LICENSE [hunk 1]
```diff
@@ -0,0 +1,674 @@
...
+
+    a) The work must carry prominent notices stating that you modified
+    it, and giving a relevant date.
+
+    b) The work must carry prominent notices stating that it is
+    released under this License and any conditions added under section
+    7.  This requirement modifies the requirement in section 4 to
+    "keep intact all notices".
+
+    c) You must license the entire work, as a whole, under this
```
Comment (fe-wickham) [inline device_tree_overlays/LICENSE:222]: Given this, I don't believe GPL is compatible with our MIT license

File: device_tree_overlays/LICENSE [hunk 2]
```diff
@@ -0,0 +1,674 @@
...
+
+  A compilation of a covered work with other separate and independent
+works, which are not by their nature extensions of the covered work,
+and which are not combined with it such as to form a larger program,
+in or on a volume of a storage or distribution medium, is called an
+"aggregate" if the compilation and its resulting copyright are not
+used to limit the access or legal rights of the compilation's users
+beyond what the individual works permit.  Inclusion of a covered work
+in an aggregate does not cause this License to apply to the other
+parts of the aggregate.
```
Comment (fe-wickham) [inline device_tree_overlays/LICENSE:243]: This is what allows device drivers to use GPL while other things being offered, they are just an aggregate. Whereas when we use dto, it is just some other software being included, it is used to "form a larger program".

**Comment by fe-wickham** [comment]
> dtogen was taken from the experimentDTO branch of the utils repo. @fe-wickham if that isn't the branch we should be using, we can update this.

I would just use whats in dev. Ultimately I think we can come up with a cleaner solution than what is on experimentalDTO.

File: device_tree_overlays/LICENSE (hunk 2 above)
Comment (audiologic-vannoy) [reply device_tree_overlays/LICENSE:243]: I'm not a lawyer, but it I agree. It seems that licensing dtogen under GPL and including it with the rest of the FOST generators (which can be argued to form a larger program) would require all of the FOST generators to be license GPL. 

Might as well just change this license to MIT. It seems like we might be able to use LGPL and MIT in the same codebase, but I don't think it's worthwhile.

**Comment by audiologic-vannoy** [comment]
> > dtogen was taken from the experimentDTO branch of the utils repo. @fe-wickham if that isn't the branch we should be using, we can update this.
> 
> I would just use whats in dev. Ultimately I think we can come up with a cleaner solution than what is on experimentalDTO.

Okay, sounds good. I'll close this PR and make a new one

**Pull request closed without merging**